editor/src/core/shared/element-path-tree.ts

```diff
@@ -0,0 +1,113 @@
+import { ElementPath, ElementPathPart } from './project-file-types'
+import * as EP from './element-path'
+import { fastForEach } from './utils'
+
+export interface ElementPathTree {
+  path: ElementPath
+  children: ElementPathTreeRoot
+}
+
+function emptyElementPathTree(path: ElementPath): ElementPathTree {
+  return {
+    path: path,
+    children: [],
+  }
+}
+
+export type ElementPathTreeRoot = Array<ElementPathTree>
+
+export function buildTree(elementPaths: Array<ElementPath>): ElementPathTreeRoot {
```

Comment:
This function has replicated the behaviour of the code it has optimised, but there's a huge caveat here that we need to be careful of. This function (and the code it replaces) doesn't guarantee the correct ordering of the tree, so e.g. for the jsx:
```
<div data-uid='aaa'>
  <div data-uid='bbb'/>
  <div data-uid='ccc'/>
</div>
```
both this function and the previous code can return the children paths in the order `[ 'aaa/ccc', 'aaa/bbb' ]` (depending on what interactions in the editor lead to this point). So with that in mind, please can you either add a comment to that effect, or name the function something that clarifies the order isn't guaranteed.

Reply:
Added caveat note in 2ad32980.